app/src/main/java/com/example/moviedb/ui/main/MainViewModel.kt

```diff
@@ -3,21 +3,26 @@ package com.example.moviedb.ui.main
 import androidx.lifecycle.MutableLiveData
 import com.example.moviedb.data.model.Genre
 import com.example.moviedb.data.repository.GenreRepository
+import com.example.moviedb.data.service.api.ApiService
+import com.example.moviedb.data.service.api.RetrofitInstance
 import com.example.moviedb.ui.base.BaseViewModel
 import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
 import io.reactivex.schedulers.Schedulers
 
 class MainViewModel : BaseViewModel() {
     val mutableLiveData = MutableLiveData<List<Genre>>()
-    val repository = GenreRepository.getInstance()
+    private val repository = GenreRepository.getInstance(RetrofitInstance.getInstance()!!.create(ApiService::class.java))
+    private val compositeDisposable = CompositeDisposable()
     fun getGenres() {
-        val disposables = repository.getGenres()
+        val disposables = repository!!.getGenres()
```

Comment:
không nên dùng !! 
Kotlin sinh ra để tránh NullPointerException 
nếu viết ntn, nó sẽ sinh ra NullPointerException khi null 
// anh Trung comment that :3 
```suggestion
        repository ?: return // cach 1
        val disposables = repository?.getGenres() // cach 2
```